nvflare/private/fed/server/fed_server.py

```diff
@@ -475,6 +475,18 @@ def register_client(self, request: Message) -> Message:
             if error is not None:
                 return make_cellnet_reply(rc=F3ReturnCode.COMM_ERROR, error=error)
 
+            data = fobs.loads(request.payload)
+            shared_fl_ctx = data.get_header(ServerCommandKey.PEER_FL_CONTEXT)
+            fl_ctx.set_peer_context(shared_fl_ctx)
+
+            client_register_data = request.get_header(CellMessageHeaderKeys.CLIENT_REGISTER_DATA, {})
+            fl_ctx.set_prop(FLContextKey.CLIENT_REGISTER_DATA, client_register_data)
+            self.engine.fire_event(EventType.CLIENT_REGISTERED, fl_ctx=fl_ctx)
+
+            unauthenticated = fl_ctx.get_prop(FLContextKey.UNAUTHENTICATED)
```

Comment:
It would be better if we could also support some defined exception such as NotAuthenticated. This will make it easier for users to write the event handler to block the user. Instead of setting the UNAUTHENTICATED prop into the ctx, they can simply "raise NotAuthenticated(...)".

Reply:
Added NotAuthenticated in the fl_exception.py.  Component implementation can raise NotAuthenticated if needed.